### DISCUSSION TITLE
Edge Functions Memory Limits

### DISCUSSION THREAD

**Discussion by maximilian-lindsey** [supabase/supabase#6834] [Questions] [answered]
I created a Deno Edge Function that works perfectly well inside the Supabase Docker environment on my Mac, but whenever I execute it in production / the cloud I run into this error after a couple of seconds of execution time:

````
502: Bad Gateway (DEPLOYMENT_FAILED)

The deployment failed while serving the request.
````

My guess is that I'm running into a memory limit. According to my local logs (Supabase doesn't seem to get the logs when I run it on their infra) **my script needs around a 150 MB in memory**.

Does anyone here know **what the production memory limits** are for Supabase functions?

I can't find anything about that in the docs nor on the pricing page.

The file size of my script is 33 KB when I bundle the function via `deno bunlde`, so it shouldn't be the script size limit, that's why I'm guessing it's a memory issue.

**Comment by candymandev** [comment]
Check the logs in Supabase (under functions -> <your-function>) for the function. That may give you more info about what's going wrong. The 502 response is not too specific.

**Comment by Ziinc** [comment, marked as answer]
Deno deploy is the underlying functions engine and has a memory limit of 512mb,so you are in the clear.

See https://deno.com/deploy/docs/pricing-and-limits

As @candymandev has mentioned, try checking your function logs to see if there are execution errors.
